### DISCUSSION TITLE
Listen for early connection close

### DISCUSSION THREAD

**Discussion by exoRift** [async-graphql/async-graphql#1472] [Q&A]
How would I listen for when the connection is closed early? (AbortController use on frontend)

**Comment by noehoro** [comment]
I'd like to know the answer to this as well.

**Comment by Miaxos** [comment]
It would depend on the integration you are using to expose async-graphql, for instance with axum (or anything built on top of `tower`) you could do something like this:

```rust

struct AbortRunningRequest {}

impl Drop for AbortRunningRequest {
    fn drop(&mut self) {
      // ...
    }
}

let service = ServiceBuilder::new()
    .service_fn(move |req: Request<Body>| async move {
        
        /// When the request is cancelled, this struct will be drop which will send a
        /// cancel signal to stop the processing.
        let _abort = AbortRunningRequest {};

        // ...
    });
```

(It would not really indicate the end of the connection, but the end of the handling of the service by tower, if you really need to know when it's the end of the connection you need to deal with `hyper` directly)
